File: Baekjoon/Brute Force/마인크래프트(18111)_2.py
import sys
from collections import Counter
input = sys.stdin.readline

N, M, B = map(int, input().split())

land = []
for n in range(N):
  land += list(map(int, input().split()))
sum_height = sum(land)

land = Counter(land)
min_height = min(land)
max_height = max(land)

time = sys.maxsize
height = 0
for h in range(min_height, max_height + 1):
  if sum_height + B < N * M * h: 
    break

  time_calc = 0
  for i in land:
    if i < h:
      time_calc += (h - i) * land[i]
    if i > h:
      time_calc += 2 * (i - h) * land[i]
    
  if time_calc <= time:
    time = time_calc
    if h > height:
      height = h

print(time, height)

File: Baekjoon/Brute Force/마인크래프트(18111)_2.js

const rl = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

const input = [];

rl.on('line', (line) => {
    input.push(line);
});

rl.on('close', () => {
    const [N, M, B] = input[0].split(' ').map(Number);

    let [min, max, sum] = [256, 0, 0];
    const land = new Map();
    for (let n=0; n<N; n++) {
        input[n+1].split(' ').forEach((h) => {
            const height = +h;

            if (height < min) min = height;
            if (height > max) max = height;
            sum += height;

            if (land.has(height)) land.set(height, land.get(height) + 1);
            else land.set(height, 1);
        });
    }

    const answer = {
        time: Number.MAX_SAFE_INTEGER,
        height: 0
    }
    for (let h=min; h<max+1; h++) {
        if (sum + B < h * N * M) break;

        let time = 0;
        const entries = land.entries();
        for (const entry of entries) {
            const [height, count] = entry;

            if (height < h) time += (h - height) * count;
            if (height > h) time += 2 * (height - h) * count;
        }

        if (time <= answer.time) {
            answer.time = time;
            if (h > answer.height) answer.height = h;
        }
    }

    console.log(answer);
});